asyncapi-cli/src/main/java/io/ballerina/asyncapi/cmd/AsyncApiCmd.java

```diff
@@ -124,6 +220,146 @@ public void execute() {
+        asyncApiConverter.generateAsyncAPIDefinitionsAllService(balFilePath, targetOutputPath, service,
+                generatedFileType);
+
+        errors.addAll(asyncApiConverter.getErrors());
+        if (!errors.isEmpty()) {
+            for (AsyncApiConverterDiagnostic error : errors) {
+                if (error instanceof ExceptionDiagnostic exceptionDiagnostic) {
+                    AsyncAPIDiagnostic diagnostic = CmdUtils.constructAsyncAPIDiagnostic(exceptionDiagnostic.getCode(),
+                            exceptionDiagnostic.getMessage(), exceptionDiagnostic.getDiagnosticSeverity(),
+                            exceptionDiagnostic.getLocation().orElse(null));
+                    outStream.println(diagnostic);
+                    exitError(this.exitWhenFinish);
+                } else if (error instanceof IncompatibleRemoteDiagnostic incompatibleError) {
+                    AsyncAPIDiagnostic diagnostic = CmdUtils.constructAsyncAPIDiagnostic(incompatibleError.getCode(),
+                            incompatibleError.getMessage(), incompatibleError.getDiagnosticSeverity(),
+                            incompatibleError.getLocation().get());
+                    outStream.println(diagnostic);
+                }
+            }
+        }
+    }
+
+    private void asyncApiToBallerinaWs(String fileName) throws IOException {
+        AsyncApiToBallerinaGenerator generator = new AsyncApiToBallerinaGenerator();
+        generator.setLicenseHeader(this.setLicenseHeaderWs());
+        generator.setIncludeTestFiles(this.includeTestFiles);
+        final File asyncApiFile = new File(fileName);
+        getTargetOutputPathWs();
+        Path resourcePath = Paths.get(asyncApiFile.getCanonicalPath());
+        generatesClientFileWs(generator, resourcePath);
+    }
+
+    /**
+     * This util is to set the license header content which is to be added at the beginning of the ballerina files.
+     */
+    private String setLicenseHeaderWs() {
+        String licenseHeader = "";
+        try {
+            if (this.licenseFilePath != null && !this.licenseFilePath.isBlank()) {
+                Path filePath = Paths.get((new File(this.licenseFilePath).getCanonicalPath()));
+                licenseHeader = Files.readString(Paths.get(filePath.toString()));
+                if (!licenseHeader.endsWith(LINE_SEPARATOR)) {
+                    licenseHeader = licenseHeader + LINE_SEPARATOR + LINE_SEPARATOR;
+                } else if (!licenseHeader.endsWith(LINE_SEPARATOR + LINE_SEPARATOR)) {
+                    licenseHeader = licenseHeader + LINE_SEPARATOR;
+                }
+            }
+        } catch (IOException e) {
+            outStream.println(String.format(MESSAGE_INVALID_LICENSE_STREAM, this.licenseFilePath, e.getMessage()));
+            exitError(this.exitWhenFinish);
+        }
+        return licenseHeader;
+    }
+
+    /**
+     * This util is to get the output Path.
+     */
+    private void getTargetOutputPathWs() {
```

Comment:
The method naming here is invalid. It says this is a getter but it does not return anything. Better to review the conventions used to name these methods.